### ISSUE TITLE
Layer linter confuses layers with layer submodules

### ISSUE THREAD

**Issue by lsorber** [seddonym/layer_linter#85]
* Layer Linter version: 0.11.0
* Python version: 3.6.7
* Operating System: macOS

### What happened

My Python project has this (simplified) structure:
```
mypackage/
    api/__init__.py
        api.py
    data/__init__.py
         api.py
    config/__init__.py
           config.py
```

Note that the contents of `mypackage/data/__init__.py` looks like:
```python
from .api import some_function
__all__ = ['some_function'] 
```

Since I would like to impose the hierarchy `mypackage.api > mypackage.data > mypackage.config`, I wrote a `layers.yml`:
```yml
Simple Contract:
  containers:
    - mypackage
  layers:
    - api
    - data
    - config
```

### What went wrong

When I run `layer-lint mypackage`, layer-lint complains that:
```
1. mypackage.data imports mypackage.api:
```

### What I expected

I would not expect the contract to be broken: the contract requires that `mypackage.data` does not import the package `mypackage.api`. Here, `mypackage.data` imports it's own private module `mypackage.data.api` which should be legal under the contract.

Also, I noticed the documentation says that:
> Remember, a Python module can either be a `.py` file or a directory with an `__init__.py` file inside.

In my experience, a `.py` file is referred to as a module, while a directory with an `__init__.py` is referred to as a package.

**Comment by seddonym** [comment]
Thanks very much for the excellent bug report.

I've been able to reproduce the bug: the problem is that Layer Linter is interpreting relative imports within an `__init__.py` as at the level of the containing directory, not the sibling modules.

I'll aim to fix this bug in the near future.

In the meantime, you may be interested in [Grimp](https://github.com/seddonym/grimp/). It offers a Python API for import analysis, and doesn't suffer from this bug. Layer Linter will soon use Grimp to do static analysis, which should fix this bug.

**Comment by seddonym** [comment]
This should now be fixed, in version 0.12.0.
